Remove default bos/eos from metadata

lucylq opened the pull request:
Summary:
Also see: https://github.com/pytorch/executorch/pull/15215

Currently:
- default eos/bos tokens are embedded into the pte
- llama3 instruct has a different set of eos/bos tokens
- users must manually specify at export time the llama3 instruct eos/bos tokens, because the runner overrides tokenizer eos/bos with the values in the PTE

This diff:
- removes the defaults
- rely on tokenizer for eos/bos UNLESS the user explicitly specifies in the metadata, in which case use the eos/bos saved in PTE.

Differential Revision: D84942718

jackzhxng reviewed (approved):
Glad to get rid of fairseq

lucylq committed:
Remove default bos/eos from metadata (#15231)

Summary:

See: https://github.com/pytorch/executorch/pull/15215

Currently:
- default eos/bos tokens are embedded into the pte
- llama3 instruct has a different set of eos/bos tokens
- users must manually specify at export time the llama3 instruct eos/bos tokens, because the runner overrides tokenizer eos/bos with the values in the PTE

This diff:
- removes the defaults
- rely on tokenizer for eos/bos UNLESS the user explicitly specifies in the metadata, in which case use the eos/bos saved in PTE.

Reviewed By: jackzhxng

Differential Revision: D84942718
